.taskcluster.yml

```diff
@@ -67,11 +68,134 @@ tasks:
           && git fetch origin --tags
           && git config advice.detachedHead false
           && git checkout {{ event.version }}
-          && ./gradlew --no-daemon clean test detektCheck ktlint assembleRelease docs lint
+          && ./gradlew --no-daemon clean test detektCheck ktlint assembleRelease docs lint uploadArchives zipMavenArtifacts
           && python automation/taskcluster/release/fetch-bintray-api-key.py
           && ./gradlew bintrayUpload --debug
       features:
         taskclusterProxy: true
+
+      artifacts:
+        public/build/lib.dataprotect.maven.zip:
```

Comment:
I assume there's isn't a way to iterate through all modules? We add new components frequently. Then we wouldn't have to remember adding them here?

Reply:
After talking to @MihaiTabara, this is a temporary solution while #803 is addressed. This will become more dynamic once we have what we call a "decision task".

The release task will become what a "decision task". In the context of Taskcluster, a decision task is what spawns a task group. The code of this decision task is stored in the components' github repo. We'll create a task group that contains the build+bintray task, then 1 "upload to maven.m.o" (aka beetmover) task per component. This way, if you want to add a new component, then it's a matter of updating the decision to spawn a new "beetmover" task. So yes, there is a second where you need to define that a new component was added. We can probably infer this from the tree structure. By the way, how does gradle keep track of the components @csadilek ?

This also paves the way for parallel release builds, but that's another topic.

Reply:
@JohanLorenzo OK, it's fine to add a new component to a file when it's created. We have to do that in Gradle too. I was just wondering if there's an easier way: https://github.com/mozilla-mobile/android-components/blob/master/settings.gradle

Reply:
We could probably parse this file with Python. It doesn't seem too complex. What do you think @MihaiTabara ?

Reply:
SGTM 👍 With the artifacts paths being so predictive, most likely we can parse the aforementioned file and:

- process the artifacts list in the "build+bintray" task
- use name + artifact to generically ramp-up another beetmover task per each of the artifacts.

Will have more info once I'm glancing better at the artifacts list (number per module, naming, etc).I'm still working on making the decision task ramp-up the build + beetmover task in my https://github.com/mozilla-mobile/android-components/pull/803

 the artifacts list within the decision task.

Reply:
I'm currently tackling this problem in https://github.com/mozilla-mobile/android-components/pull/803 and after speaking with @csadilek, sounds like it might be too much effort in the end. 

If all the modules published their artifacts, we could've easily parse that `settings.gradle` file and generate that generic artifact entry. But since we have modules that don't publish, this gets slightly tricky as we need some flags to know which one is which, but we're not allowed to touch the file except maybe with comments. Overall, this beats the purpose and adds too much complexity vs. adding a new line in `.taskcluster.yml` each time a new module is added.

Agreed with @csadilek to re-evaluate in a near future, based on the future added modules, how/if problematic this will be and if it's worth optimizing. 

CC: @JohanLorenzo